compiler/src/iree/compiler/Codegen/Common/ReconcileTranslationInfo.cpp

```diff
@@ -364,6 +374,191 @@ resolveWorkgroupForAll(RewriterBase &rewriter, FunctionOpInterface funcOp,
+///   `%orig_numworkgroups` is the number of workgroups along `k` that were used
+///   before the resolution of the split reduction `scf.forall`. This is same as
+///   `hal.interface.workgroup.count[k] / %nsplit`.
+/// - All uses of `%iv` is replaced by `hal.interface.workgroup.id[k] / %nsplit`
+/// ```
+///
+static LogicalResult
+resolveSplitReduceForAll(RewriterBase &rewriter, FunctionOpInterface funcOp,
+                         IREE::Codegen::WorkgroupId delinearizeFrom) {
+  Region &body = funcOp.getFunctionBody();
+  if (body.empty()) {
+    return success();
+  }
+
+  SmallVector<scf::ForallOp> splitReductionForAllOps;
+  funcOp.walk([&splitReductionForAllOps](scf::ForallOp forAllOp) {
+    auto mapping = forAllOp.getMapping();
+    if (!mapping || mapping->size() != 1 ||
+        !isa<IREE::LinalgExt::SplitReductionMappingAttr>(
+            mapping->getValue().front())) {
+      return;
+    }
+    splitReductionForAllOps.push_back(forAllOp);
+  });
+
+  if (splitReductionForAllOps.empty()) {
+    return lowerSplitReductionModifierOp(rewriter, funcOp, delinearizeFrom,
+                                         rewriter.getIndexAttr(1));
+  }
+
+  // For now support only a single split-reduction forall.
+  if (splitReductionForAllOps.size() != 1) {
+    return funcOp->emitOpError(
+        "failed to resolve multiple split-reduction loops");
+  }
+
+  scf::ForallOp forallOp = splitReductionForAllOps.front();
+  if (forallOp->getNumResults() != 0) {
+    return forallOp.emitOpError(
+        "cannot resolve for all ops with return values");
+  }
+
+  assert(forallOp.getMapping() && forallOp.getMapping()->size() == 1 &&
+         "expected forall op to have mapping of "
+         "`[#iree_codegen.split_reduction_mapping]`");
+  // Currently only support single induction variable.
+  if (forallOp.getRank() != 1) {
+    return forallOp->emitOpError("unsupported split-reduction resolution of "
+                                 "`scf.forall` op of rank greater than 1");
+  }
+
+  OpBuilder::InsertionGuard g(rewriter);
+  rewriter.setInsertionPoint(forallOp);
+  Location loc = forallOp.getLoc();
+
+  OpFoldResult lb = forallOp.getMixedLowerBound().front();
+  OpFoldResult ub = forallOp.getMixedUpperBound().front();
+  OpFoldResult step = forallOp.getMixedStep().front();
+  AffineExpr s0, s1, s2;
+  bindSymbols(rewriter.getContext(), s0, s1, s2);
+  OpFoldResult nSplitProcs = affine::makeComposedFoldedAffineApply(
+      rewriter, loc, (s1 - s0).ceilDiv(s2), {lb, ub, step});
+
+  // Lower the splitk-modifier op in the entry point.
+  if (failed(lowerSplitReductionModifierOp(rewriter, funcOp, delinearizeFrom,
+                                           nSplitProcs))) {
+    return forallOp->emitOpError("failed to lower split reduction modifier op");
+  }
+
+  auto procIdOp = rewriter.create<IREE::HAL::InterfaceWorkgroupIDOp>(
+      loc, static_cast<uint>(delinearizeFrom));
```

Comment:
Same here and below, use `to_underlying`
```suggestion
      loc, llvm::to_underlying(delinearizeFrom));
```

Reply:
undeerlying is uint64_t. I use static cast here to avoid some warnings sometimes due to implicit truncation.

Reply:
This broke builds on MSVC: https://github.com/iree-org/iree/actions/runs/16247426953/job/45872332414#step:9:6864

```
C:\home\runner\_work\iree\iree\compiler\src\iree\compiler\Codegen\Common\ReconcileTranslationInfo.cpp(526): error C2061: syntax error: identifier 'uint'
C:\home\runner\_work\iree\iree\compiler\src\iree\compiler\Codegen\Common\ReconcileTranslationInfo.cpp(528): error C2061: syntax error: identifier 'uint'
C:\home\runner\_work\iree\iree\compiler\src\iree\compiler\Codegen\Common\ReconcileTranslationInfo.cpp(529): error C3536: 'nTotalProcsOp': cannot be used before it is initialized
C:\home\runner\_work\iree\iree\compiler\src\iree\compiler\Codegen\Common\ReconcileTranslationInfo.cpp(533): error C3536: 'procIdOp': cannot be used before it is initialized
```

`uint` is not a portable type.